include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge.h

```diff
@@ -151,18 +141,25 @@ struct __parallel_merge_submitter<_IdType, __internal::__optional_kernel_name<_N
     auto
     operator()(_ExecutionPolicy&& __exec, _Range1&& __rng1, _Range2&& __rng2, _Range3&& __rng3, _Compare __comp) const
     {
-        using _Size = oneapi::dpl::__internal::__difference_t<_Range3>;
-
-        auto __n1 = __rng1.size();
-        auto __n2 = __rng2.size();
+        const _IdType __n1 = __rng1.size();
+        const _IdType __n2 = __rng2.size();
+        const _IdType __n = __n1 + __n2;
 
         assert(__n1 > 0 || __n2 > 0);
 
         _PRINT_INFO_IN_DEBUG_MODE(__exec);
 
-        const std::uint8_t __chunk = __exec.queue().get_device().is_cpu() ? 128 : 4;
-        const _Size __n = __n1 + __n2;
-        const _Size __steps = oneapi::dpl::__internal::__dpl_ceiling_div(__n, __chunk);
+        // Empirical number of values to process per work-item
+        std::uint16_t __chunk = 128;
+        if (__exec.queue().get_device().is_gpu())
+        {
+            if (__n >= 16'777'216)
```

Comment:
I am afraid the change make performance worse on some data sizes? especially > 16M:
Before:
```
4,19.737000
8,13.623000
16,13.939000
32,15.304000
64,16.719000
128,21.427000
256,16.178000
512,23.896000
1024,16.011000
2048,17.129000
4096,20.501000
8192,18.519000
16384,18.262000
32768,20.101000
65536,20.013000
131072,23.908000
262144,32.386000
524288,45.397000
1048576,76.331000
2097152,125.091000
4194304,243.039000
8388608,508.584000
16777216,1072.780000
33554432,2256.281000
67108864,4802.148000
134217728,10768.003000

```  
after:
4,37.869000
8,16.068000
16,19.697000
32,16.614000
64,17.873000
128,16.984000
256,21.193000
512,13.839000
1024,13.802000
2048,13.854000
4096,15.411000
8192,16.091000
16384,15.023000
32768,16.290000
65536,16.829000
131072,19.409000
262144,27.462000
524288,39.672000
1048576,65.975000
2097152,119.436000
4194304,235.375000
8388608,1365.266000
16777216,2108.173000
33554432,3594.338000
67108864,6973.669000
134217728,13405.574000